### DISCUSSION TITLE
is mapstruct support jdk 17 ?

### DISCUSSION THREAD

**Discussion by MostafaSwelam** [mapstruct/mapstruct#2886] [General]
i wonder if any version of mapstruct support jdk 17

**Comment by filiphr** [comment]
Yes MapStruct does support JDK 17. We have tested 1.4 and 1.5 with Java 17, perhaps older versions work as well, but they have not been tested.

**Reply by famod** [reply]
I can confirm that statement: We are using mapstruct 1.5 (1.4 not long ago) productively on Java 17 (we had all versions since 17.0.0, IIRC).

**Comment by georgemarsroberts** [comment]
17.0.2 can not

**Reply by filiphr** [reply]
This is incorrect. We are testing with Java 17 and everything is running green. Please provide more information about what isn't working.
